### PR TITLE
14.6.7 - Bordered pricing cards, static accordion themes, SB 8.3.5

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

The motivation lies within

## Summary of changes

A mixture of changes that benefit design requests for the pricing page and the pricing dashboards.

- for the pricing page, `static` and `darkStatic` themes for the `Accordion`. This is basically a hack to make it more like a table instead of retooling the table-ish bit of the pricing page to use a different component (`Accordion` was chosen for its expandable sections)
- for the dash, a `border` prop can be supplied to the data objects that go into the pricing card to give it a stylised border with some custom text. More on that [here](https://ably.github.io/ably-ui/?path=/story/features-pricing-cards--bordered-plans)
- for both, the description section of the pricing/consumption cards adjusts to the height of the largest one in the set so that the CTAs stay in line
- marks a couple of flaky stories as such to stop the test runner borking

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Release Notes

- **New Features**
	- Introduced new themes for the Accordion component, including `static` and `darkStatic`.
	- Added a new story for `StaticAndFullyOpen` in Accordion to showcase static themes.
	- Enhanced PricingCards with customizable borders and dynamic height management for descriptions.
	- Added a new story, `BorderedPlans`, to display pricing plans with borders.
	- Updated PricingCards to support a new `border` property for enhanced styling options.
	- Expanded support for Tailwind CSS to include JSON files for style processing.

- **Bug Fixes**
	- Improved interaction logic for Accordion based on theme selection.

- **Documentation**
	- Updated stories and documentation for Accordion and PricingCards to reflect new features and usage.

- **Chores**
	- Updated dependencies related to Storybook for improved performance and features.
	- Expanded the list of flaky stories in the test runner for better testing accuracy.
	- Updated version number in the package.json file.

### CODE REVIEW

File: src/core/Accordion.tsx [hunk 1]
```diff
@@ -61,11 +61,30 @@ const themeClasses: Record<AccordionTheme, AccordionThemeColors> = {
...
+    selectableText: "text-white",
+  },
+  darkStatic: {
+    bg: "bg-neutral-1200",
+    hoverBg: "hover:bg-neutral-1200",
+    text: "text-white",
+    toggleIconColor: "text-neutral-1200",
+    selectableBg: "bg-neutral-1200",
+    selectableText: "text-neutral-1300",
+  },
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Accordion.tsx:79]: Good catch, but this is the intention in these scenarios.

File: src/core/Pricing/PricingCards.tsx [hunk 2]
```diff
@@ -40,130 +64,154 @@
...
+                  ))}
+                </div>
               </div>
-            </div>
-            {delimiterColumn(index)}
-          </Fragment>
-        ))}
+              {delimiterColumn(index)}
+            </Fragment>
+          ),
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Pricing/PricingCards.tsx:213]: There is no `id` property, the titles will be distinct here.

**Review by kennethkalmer — COMMENTED** [review]
This looks so nice in Chrome, and then enter Safari 😭 

Pricing Cards - Docs ⋅ Storybook 2024-10-08 20-48-23

**Review by aralovelace — CHANGES_REQUESTED** [review]
blue border is not set yet for light theme

image

File: src/core/Pricing/PricingCards.tsx [hunk 3]
```diff
@@ -17,6 +18,32 @@ const PricingCards = ({
...
+
+    determineMaxDescriptionHeight();
+
+    addEventListener("resize", determineMaxDescriptionHeight);
+
+    return () => {
+      removeEventListener("resize", determineMaxDescriptionHeight);
+      determineMaxDescriptionHeight.cancel();
+    };
+  }, [descriptionsRef.current]);
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Pricing/PricingCards.tsx:45]: Good bunny

File: src/core/Pricing/PricingCards.tsx [hunk 4]
```diff
@@ -17,6 +18,32 @@ const PricingCards = ({
...
+  useEffect(() => {
+    determineMaxDescriptionHeight();
+
+    window.addEventListener("resize", determineMaxDescriptionHeight);
+
+    return () => {
+      window.removeEventListener("resize", determineMaxDescriptionHeight);
+      determineMaxDescriptionHeight.cancel();
+    };
+  }, []);
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Pricing/PricingCards.tsx:45]: Decent suggestions, but this is a static component with very few re-render cycles so the memoisation would appear to me to be an overoptimisation here

**Review by kennethkalmer — APPROVED** [review]
I'm happy with the Safari compatibility now! Maybe hold for an additional ✅ from @aralovelace to make sure she's happy as well?

**Review by aralovelace — APPROVED** [review]
⭐⭐⭐⭐⭐

**Pull request merged**